public/app/features/integrated-alerting/components/NotificationChannel/AddNotificationChannelModal/AddNotificationChannelModal.tsx

```diff
@@ -37,18 +66,25 @@ export const AddNotificationChannelModal: FC<AddNotificationChannelModalProps> =
       logger.error(e);
     }
   };
-  const renderTypeFields = useCallback((values: NotificationChannelRenderProps) => {
-    const TypeFields = TYPE_FIELDS_COMPONENT[values.type.value];
 
-    return <TypeFields values={values} />;
-  }, []);
+  const mutators: Record<
+    MutatorKeys,
+    Mutator<NotificationChannelRenderProps, Partial<NotificationChannelRenderProps>>
+  > = {
+    // Insead of just resetting the value to '', we'll use the initial value
+    // This way, it keeps working on edition mode
+    resetKey: ([key]: PagerDutyKeyType[], state, utils) => {
+      utils.changeValue(state, key, () => initialValues[key]);
```

Comment:
What is the initial value for pager duty service/routing?

Reply:
If you're adding a new notification channel, it'll be "". If editing one, Then it's the value already there. These values are coming from `getInitialValues()` on the utils.

Reply:
If you're adding a new notification channel, it'll be "". If editing one, Then it's the value already there. These values are coming from getInitialValues() on the utils.

Reply:
Got it, thanks Fabio!